### PR TITLE
feat: add first version of dogfooding app

### LINKED ISSUE #69: feat: Create first version of Waku dogfooding app
This is a **feature request**

## Problem

Current efforts of testing long-term usage of js-waku required manually running an example and observing it. We should move towards a more automated solution. Additionally, it's important to gather metrics for how js-waku behaves in realistic conditions, across many browsers and locations.

## Proposed Solutions

Create a webapp that runs a waku node and sends/listens to sequences of messages. Every time a message is successfully sent or received, it sends a request to the [telemetry service](https://github.com/waku-org/telemetry) with relevant metrics along with a wallet address.

To complete sequences:
- generate random number and hash it
- randomly pick a number of total messages in the sequence, between 5 and 20
- randomly pick how often to send each message, between 1 and 3 seconds
- send structured messages starting with index 0 to the total number using light push
- each time a message is successfully sent to at least one peer, push the information to the telemetry service

Similarly, listen for messages using filter and push information to the telemetry service whenever one is decoded and received.

## Notes

- Related work to update telemetry service to work with this application: https://github.com/waku-org/telemetry/pull/1

### PR DESCRIPTION
**Pull request by adklempner**
Resolves #69

### CODE REVIEW

File: examples/sent-received-ratio/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            payload: utf8ToBytes(String(i)),
+            timestamp: new Date(),
+          });
+          controller.enqueue({ result, i });
+          i++;
+          if (i >= numMessages) {
+            clearInterval(intervalId);
+            controller.close();
+          }
+        }, period);
```
[the remark this answers is not in the document]
Comment (adklempner) [reply examples/sent-received-ratio/src/index.ts]: good catch, fixed

File: examples/sent-received-ratio/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    defaultBootstrap: true,
+  });
+  await node.start();
+  await waitForRemotePeer(node);
+  const encoder = createEncoder({ contentTopic: DEFAULT_CONTENT_TOPIC });
+
+  const start = (numMessages: number, period: number = 3000) => {
+    let intervalId: NodeJS.Timeout;
+    let i = 0;
+    const stream = new ReadableStream({
```
[the remark this answers is not in the document]
Comment (adklempner) [reply examples/sent-received-ratio/src/index.ts]: removed since scope has changed

**Review by weboko — APPROVED** [review]

File: examples/dogfooding/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        message.payload
+      );
+
+      // Don't bother reporting messages sent by this same node
+      if (sequencedMessage.sender === peerId) {
+        return;
+      }
+      telemetryClient.push<TelemetryPushFilter>([
+        {
+          messageType: TelemetryType.LIGHT_PUSH_FILTER,
```
Comment (weboko) [inline examples/dogfooding/src/index.ts:133]: hm. is it `lightPush` or `filter`? 
if first - I read it as `sent by lightPush`
if second - `received by filter`

Comment (adklempner) [reply examples/dogfooding/src/index.ts:133]: It's both. If Sender and Reporter are the same, it's considered sent via light push. If different, then received via filter.

Comment (weboko) [reply examples/dogfooding/src/index.ts:133]: so, we can avoid this altogether?

**Review by weboko — APPROVED** [review]

File: examples/dogfooding/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  });
+};
+
+export async function app(telemetryClient: TelemetryClient) {
+  const node = await wakuNode();
+  await node.start();
+
+  // TODO: https://github.com/waku-org/js-waku/issues/2079
+  // Dialing bootstrap peers right on start in order to have Filter subscription initiated properly
+  await node.dial("/dns4/node-01.do-ams3.waku.test.status.im/tcp/8000/wss");
```
Comment (weboko) [inline examples/dogfooding/src/index.ts:44]: Mitigation for https://github.com/waku-org/js-waku/issues/2079

File: examples/dogfooding/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        const payload = ProtoSequencedMessage.encode(message).finish();
+        const result = await node.lightPush.send(encoder, {
+          payload,
+          timestamp: new Date(),
+        });
+        console.log("light push successes: ", result.successes.length);
+        console.log("light push failures: ", result.failures.length);
+        if (result.successes.length > 0) {
+          // Push to telemetry client
+          telemetryClient.push<TelemetryPushFilter>([
```
Comment (weboko) [inline examples/dogfooding/src/index.ts:80]: We need to report failures with a message of failure. 
What columns can be used for that @adklempner ?

**Pull request merged**